ledger/apply/apply.go

```diff
@@ -18,12 +18,23 @@ package apply
 
 import (
 	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto/stateproof"
 	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/data/stateproofmsg"
 	"github.com/algorand/go-algorand/data/transactions"
 	"github.com/algorand/go-algorand/data/transactions/logic"
 	"github.com/algorand/go-algorand/ledger/ledgercore"
 )
 
+// StateProofs allows fetching and updating state-proofs state on the ledger
+type StateProofs interface {
+	BlockHdr(r basics.Round) (bookkeeping.BlockHeader, error)
+	StateProofNext() basics.Round
+	SetStateProofNext(rnd basics.Round)
+	ValidateStateProof(latestRoundInIntervalHdr *bookkeeping.BlockHeader, stateProof *stateproof.StateProof, votersHdr *bookkeeping.BlockHeader, nextStateProofRnd basics.Round, atRound basics.Round, msg *stateproofmsg.Message) error
```

Comment:
I recommend moving this to the side of the caller / internal ... ValidateStateProof doesn't do anything with the ledger